components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/TemplateResource.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.micro.integrator.management.apis;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.endpoints.Template;
+import org.apache.synapse.mediators.template.TemplateMediator;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.inbound.endpoint.internal.http.api.APIResource;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents template resources defined in the synapse configuration.
+ **/
+public class TemplateResource extends APIResource {
+
+    private static final Log log = LogFactory.getLog(TemplateResource.class);
+
+    /* Name of the JSON list of endpoint templates */
+    private static final String ENDPOINT_TEMPLATE_LIST = "endpointTemplateList";
+    /* Name of the JSON list of sequence templates */
+    private static final String SEQUENCE_TEMPLATE_LIST = "sequenceTemplateList";
+    /* Template type parameter */
+    private static final String TEMPLATE_TYPE_PARAM = "templateType";
```

Comment:
Since we have "templates" in the path param https://github.com/wso2/micro-integrator/pull/248/files#diff-834b97a73567a713ccfa74fa8f0e0a09R31 shall we make this query param name as just  **type** since it should be intuitive

Reply:
done.